xrpl_types/src/amount.rs

```diff
@@ -51,7 +43,29 @@ impl Amount {
     pub fn size(&self) -> f64 {
         match self {
             Amount::Drops(value) => value.parse::<f64>().unwrap() / 1_000_000.0,
-            Amount::Issued { value, .. } => value.parse::<f64>().unwrap(),
+            Amount::Issued(IssuedTokenAmount { value, .. }) => value.parse::<f64>().unwrap(),
+        }
+    }
+}
+
+/// Amount of issued token. See <https://xrpl.org/currency-formats.html#token-amounts>
+#[derive(Debug, Eq, PartialEq, Serialize, Deserialize, Clone)]
+pub struct IssuedTokenAmount {
```

Comment:
Extracted the type `IssuedTokenAmount` should that this can be used in places where amount is always a token amount (like in `RippleState`)